### DISCUSSION TITLE
Yunohost: error during script installation

### DISCUSSION THREAD

**Discussion by HerthBC** [tteck/Proxmox#1436] [Help Needed]
Hi, 
tried to install Yunohost on my Proxmox Server and always stops with an error:

Creating a YunoHost LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using local-lvm for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 108 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.1.32  
 ✓ Internet Connected
 ✓ DNS Resolved github.com to 140.82.121.3
 - Updating Container OS...
**[ERROR] in line 24: exit code 100: while executing command "$@" > /dev/null 2>&1
root@proxmox:~#** 

LXC container is running afterwards, but there is no access via browser and login on the console is not possible (wrong user/password).
Is there something I can do/try or is there a problem with the script?

Thank you,
Herth

**Comment by tteck** [comment]
Run in verbose mode to see the actual issue.

**Reply by HerthBC** [reply]
Here is the shortened verbose modus output.
Error occurs when trying to 'Installing Dependencies'


✓ Using local for Template Storage.
✓ Using local-lvm for Container Storage.
✓ Updated LXC Template List
✓ LXC Container 108 was successfully created.
✓ Started LXC Container
✓ Set up Container OS
✓ Network Connected:
✓ Internet Connected
✓ DNS Resolved github.com to 140.82.121.4
✓ Updated Container OS
+ msg_info 'Installing Dependencies'
+ local 'msg=Installing Dependencies'
+ echo -ne ' - \033[33mInstalling Dependencies...'

_….. (lots of stuff is installed without any issues, here are the last lines when the errors starts.)_

The following NEW packages will be installed:
  dirmngr gnupg gnupg-l10n gnupg-utils gpg gpg-agent gpg-wks-client gpg-wks-server gpgconf gpgsm libassuan0 libksba8 libnpth0 pinentry-curses
0 upgraded, 14 newly installed, 0 to remove and 0 not upgraded.
Need to get 7,665 kB of archives.
After this operation, 15.7 MB of additional disk space will be used.
Err:1 http://deb.debian.org/debian bullseye/main amd64 libassuan0 amd64 2.5.3-7.1
  Temporary failure resolving 'deb.debian.org'
Ign:2 http://deb.debian.org/debian bullseye/main amd64 gpgconf amd64 2.2.27-2+deb11u2
Get:2 http://deb.debian.org/debian bullseye/main amd64 gpgconf amd64 2.2.27-2+deb11u2 [548 kB]
Get:3 http://deb.debian.org/debian bullseye/main amd64 libksba8 amd64 1.5.0-3+deb11u2 [123 kB]
Get:4 http://deb.debian.org/debian bullseye/main amd64 libnpth0 amd64 1.6-3 [19.0 kB]  
Get:5 http://deb.debian.org/debian bullseye/main amd64 dirmngr amd64 2.2.27-2+deb11u2 [763 kB]
Get:6 http://deb.debian.org/debian bullseye/main amd64 gnupg-l10n all 2.2.27-2+deb11u2 [1,086 kB]
Get:7 http://deb.debian.org/debian bullseye/main amd64 gnupg-utils amd64 2.2.27-2+deb11u2 [905 kB]
Get:8 http://deb.debian.org/debian bullseye/main amd64 gpg amd64 2.2.27-2+deb11u2 [928 kB]
Get:9 http://deb.debian.org/debian bullseye/main amd64 pinentry-curses amd64 1.1.0-4 [64.9 kB]
Get:10 http://deb.debian.org/debian bullseye/main amd64 gpg-agent amd64 2.2.27-2+deb11u2 [669 kB]
Get:11 http://deb.debian.org/debian bullseye/main amd64 gpg-wks-client amd64 2.2.27-2+deb11u2 [524 kB]
Get:12 http://deb.debian.org/debian bullseye/main amd64 gpg-wks-server amd64 2.2.27-2+deb11u2 [516 kB]
Get:13 http://deb.debian.org/debian bullseye/main amd64 gpgsm amd64 2.2.27-2+deb11u2 [645 kB]
Get:14 http://deb.debian.org/debian bullseye/main amd64 gnupg all 2.2.27-2+deb11u2 [825 kB]
Fetched 7,615 kB in 1min 21s (94.3 kB/s)
E: Failed to fetch http://deb.debian.org/debian/pool/main/liba/libassuan/libassuan0_2.5.3-7.1_amd64.deb  Temporary failure resolving 'deb.debian.org'
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
++ error_handler 20 '$STD apt-get install -y gnupg'
++ local exit_code=100
++ local line_number=20
++ local 'command=$STD apt-get install -y gnupg'
++ local 'error_message=\033[01;31m[ERROR]\033[m in line \033[01;31m20\033[m: exit code \033[01;31m100\033[m: while executing command \033[33m$STD apt-get install -y gnupg\033[m'
++ echo -e '\n\033[01;31m[ERROR]\033[m in line \033[01;31m20\033[m: exit code \033[01;31m100\033[m: while executing command \033[33m$STD apt-get install -y gnupg\033[m'

[ERROR] in line 20: exit code 100: while executing command $STD apt-get install -y gnupg
++ [[ 20 -eq 22 ]]
+ exit
root@proxmox:~# 


Regards,
Herth

**Comment by tteck** [comment]
> Err:1 http://deb.debian.org/debian bullseye/main amd64 libassuan0 amd64 2.5.3-7.1
Temporary failure resolving 'deb.debian.org'

DNS issues?

**Comment by tteck** [comment]
No issues here:

```
__  __                  __  __           __ 
\ \/ /_  ______  ____  / / / /___  _____/ /_
 \  / / / / __ \/ __ \/ /_/ / __ \/ ___/ __/
 / / /_/ / / / / /_/ / __  / /_/ (__  ) /_  
/_/\__,_/_/ /_/\____/_/ /_/\____/____/\__/  
                                            
Using Advanced Settings
Using Distribution: debian
Using debian Version: 11
Using Container Type: 1
Using Root Password: Automatic Login
Container ID: 112
Using Hostname: yunohost
Using Disk Size: 20
Allocated Cores: 2
Allocated RAM: 2048
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using MAC Address: AE:1A:60:07:B7:59
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a YunoHost LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using local-lvm for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 112 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.253 
 ✓ Internet Connected
 ✓ DNS Resolved github.com to 140.82.112.4
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Installed YunoHost
 ✓ Customized Container
 ✓ Cleaned
 ✓ Completed Successfully!

YunoHost should be reachable by going to the following URL.
         http://192.168.86.253/ 
```
![Screenshot 2023-05-23 4 07 17 AM](https://github.com/tteck/Proxmox/assets/21342670/8f7b9bfd-c7e7-43cc-9c24-7bb0e04d7711)

**Reply by HerthBC** [reply]
okay, thank you for trying. I will have a look for DNS issues on my side.

Regards, Herth

**Comment by matkes** [comment]
I'm getting a similar error message:
✓ Using local for Template Storage.
✓ Using local-lvm for Container Storage.
✓ Updated LXC Template List
✓ LXC Container 108 was successfully created.
✓ Started LXC Container
✓ Set up Container OS
✓ Network Connected: 10.0.9.254
✓ IPv4 Internet Connected
✗ IPv6 Internet Not Connected
✓ DNS Resolved github.com to 140.82.121.3
✓ Updated Container OS
✓ Installed Dependencies
/ Installing YunoHost (Patience)
[ERROR] in line 23: exit code 0: while executing command "$@" > /dev/null 2>&1
The silent function has suppressed the error, run the script with verbose mode enabled, which will provide more detailed output.

Other LXC CTs from your library do install on my end with no errors.
